src/Tickets/Commerce/Gateways/Square/Application_Fee.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Handles the application fee calculations for Square transactions.
+ *
+ * This class manages the application fee calculations for Square transactions,
+ * specifically for US-based merchants. The fee is only applied to unlicensed
+ * installations and US-based sales.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square
+ */
+
+namespace TEC\Tickets\Commerce\Gateways\Square;
+
+use TEC\Tickets\Commerce\Settings;
+use TEC\Tickets\Commerce\Utils\Value;
+
+/**
+ * The Square Application_Fee class
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Square
+ */
+class Application_Fee {
+
+	/**
+	 * The percentage applied to Square transactions. Currently set at 2%.
+	 *
+	 * @since TBD
+	 *
+	 * @var float
+	 */
+	const FIXED_FEE = 0.02;
+
+	/**
+	 * Calculate the fee value that needs to be applied to the PaymentIntent.
+	 *
+	 * @since TBD
+	 *
+	 * @param Value $value the value over which to calculate the fee.
+	 *
+	 * @return Value;
+	 */
+	public static function calculate( Value $value ) {
+		if ( Settings::is_licensed_plugin( true ) ) {
+			return Value::create();
+		}
+
+		// Check if merchant is in the US - Square application fees only apply to US sales.
+		$merchant         = tribe( Merchant::class );
+		$merchant_country = $merchant->get_merchant_country();
+
+		if ( 'US' !== $merchant_country ) {
+			return Value::create();
+		}
```

Comment:
Should we change the order here ? So that non-US merchant dont even have their license validated through a remote API call ?